src/csharp/Grpc.IntegrationTesting/ServerRunners.cs

```diff
@@ -79,8 +79,7 @@ public static IServerRunner CreateStarted(ServerConfig config)
             ServerServiceDefinition service = null;
             if (config.ServerType == ServerType.AsyncServer)
             {
-                GrpcPreconditions.CheckArgument(config.PayloadConfig == null,
-                    "ServerConfig.PayloadConfig shouldn't be set for BenchmarkService based server.");    
+                Logger.Warning("ServerConfig.PayloadConfig is not supported for C# ServerType.AsyncServer. Ignoring the value");
```

Comment:
why does PayloadConfig needs to be set? For AsyncServer (= non-generic), the request should contain the payload config to use for the response, so something is wrong here.
Which PR have populated payload_config?